pkgs/by-name/ay/ayugram-desktop/package.nix

```diff
@@ -0,0 +1,60 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  telegram-desktop,
+  withWebkit ? true,
+}:
+
+telegram-desktop.override {
+  pname = "ayugram-desktop";
+  inherit withWebkit;
+  unwrapped = telegram-desktop.unwrapped.overrideAttrs (
+    finalAttrs: previousAttrs: {
+      pname = "ayugram-desktop-unwrapped";
+      version = "5.8.3";
+
+      src = fetchFromGitHub {
+        owner = "AyuGram";
+        repo = "AyuGramDesktop";
+        tag = "v${finalAttrs.version}";
+        hash = "sha256-bgfqYI77kxHmFZB6LCdLzeIFv6bfsXXJrrkbz5MD6Q0=";
+        fetchSubmodules = true;
+      };
+
+      # Since the original .desktop file is for Flatpak, we need to fix it
+      postPatch =
+        (previousAttrs.postPatch or "")
+        +
+          lib.optionalString stdenv.hostPlatform.isLinux
+            # Rudiment: related functionality is disabled by disabling the auto-updater
+            # and it breaks the .desktop file in Aylur's Gtk Shell
+            # (with it, it causes the application to not be seen by the app launcher)
+            # https://github.com/AyuGram/AyuGramDesktop/blob/5566a8ca0abe448a7f1865222b64b68ed735ee07/Telegram/SourceFiles/platform/linux/specific_linux.cpp#L455
+            ''
+              substituteInPlace lib/xdg/com.ayugram.desktop.desktop \
+                --replace-fail "DESKTOPINTEGRATION=1 " ""
+            ''
+        +
+          # Since we aren't in Flatpak, "DBusActivatable" has no unit to
```

Comment:
I'm perhaps a bit too late here but this comment isn't accurate, DBusActivatable has nothing to do with flatpak and you don't need a unit (is this referring to a systemd unit?) for it. If it fails to start with this option set, it means there are deeper problems with system integration.

Reply:
I don't know whether it runs but if it doesn't it means something works incorrectly which will lead to problems